pydantic/_internal/_generate_schema.py

```diff
@@ -1391,9 +1365,7 @@ def _type_alias_type_schema(
                 annotation = replace_types(annotation, typevars_map)
                 schema = self.generate_schema(annotation)
                 assert schema['type'] != 'definitions'
-                schema['ref'] = ref  # type: ignore
-            self.defs.definitions[ref] = schema
-            return core_schema.definition_reference_schema(ref)
+                return self.defs.create_def_ref(ref, {**schema, 'ref': ref})  # type: ignore
```

Comment:
I dont understand how things here manage to work as this previously mutated the schema dict which may have been given by existing `__pydantic_core_schema__` in a model. So this went and possibly mutated things incorrectly. The generation logic looks quite dangerous in many places.